### PR TITLE
Flight of stairs in java (Dp)

### PR DESCRIPTION
**Pull request by FireQueen-3010**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

The flight of stairs problem using dp in Java.

Given two positive integers : n and k. We are currently at stair 0, and want to climb to stair n. We can either climb 1, 2 or 3 stairs at a time, but we can climb 3 stairs only k times.  Return the number of ways to climb to the nth stair. 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#4223 (java)

### CODE REVIEW

**Comment by FireQueen-3010** [comment]
Please review my pr @HarshCasper @siddharth25pandey @ankitaggarwal23

File: Java/dp/FlightOfStairs.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/*Problem link: https://binarysearch.com/problems/A-Flight-of-Stairs-Sequel
+You are given positive integers n and k. You are currently at stair 0 and want to climb to stair n. You can either climb 1, 2, or 3 stairs at a time, but you can only climb 3 stairs at most k times.
+
+Return the number of ways to climb to the nth stair. The answer is guaranteed to fit in a 32-bit signed integer.
+*/
```
Comment (HarshCasper) [inline Java/dp/FlightOfStairs.java]: Try to modularize this content. Make it short, crisp and readable

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by FireQueen-3010** [comment]
Done with the changes , @HarshCasper . Please review :)

File: Java/dp/FlightOfStairs.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            memo[n][k] += ans(memo, n - 3, k - 1);
+
+       /*returning memoized answer*/
+        return memo[n][k];
+
+    }
+    public static void main(String args[]) {
+        Scanner sc = new Scanner(System.in);
+        int n = sc.nextInt();
+        int k = sc.nextInt();
```
Comment (ankitaggarwal23) [inline Java/dp/FlightOfStairs.java]: Add a display message.

File: Java/dp/FlightOfStairs.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+/*
+Sample Input :
+n = 4
+k = 1
+Output :
+7
+Time complexity: O(n*k) 
+Space complexity: O(n*k)
+ */
```
Comment (ankitaggarwal23) [inline Java/dp/FlightOfStairs.java]: Add a proper display message
Add a new line at the end

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by FireQueen-3010** [comment]
Done with the changes please review @ankitaggarwal23  @HarshCasper  :)

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**